pam/integration-tests/helpers_test.go

```diff
@@ -148,8 +134,7 @@ func sharedAuthd(t *testing.T, args ...testutils.DaemonOption) (socketPath strin
 	args = append(slices.Clone(args), testutils.WithSharedDaemon(true))
 	sa.gPasswdOutputPath = filepath.Join(t.TempDir(), "gpasswd.output")
 	sa.groupsFile = filepath.Join(testutils.TestFamilyPath(t), "gpasswd.group")
-	sa.socketPath, sa.cleanup = runAuthdForTesting(t, sa.gPasswdOutputPath,
-		sa.groupsFile, true, useSharedInstance, args...)
+	sa.socketPath = runAuthdForTesting(t, sa.gPasswdOutputPath, sa.groupsFile, true, useSharedInstance, args...)
```

Comment:
So the problem that we have here is that the test that actually started the test may not be the one finishing it, and so we'd end up removing the socket path or stopping it too early, reason why for the complication.

So I'm fine for the change in principle, but I would keep a version that returns the claeanup function for this specific case.